src/saml2/mdstore.py

```diff
@@ -903,8 +923,17 @@ def load(self, *args, **kwargs):
             key = args[1]
             _md = MetaDataLoader(self.attrc, args[1], **_args)
         elif typ == "mdq":
-            key = args[1]
-            _md = MetaDataMDX(args[1])
+            if 'url' in kwargs:
+                key = kwargs['url']
```

Comment:
this is leftover

Reply:
No, key is used later after the end of the if/elif/else clause. The new code for MDQ follows the existing pattern as the 'inline', 'remote', 'mdfile', and 'loader' logic paths and sets key so that it can be used later as key in self.metadata[key].